pkg/oc/admin/router/router.go

```diff
@@ -728,11 +779,44 @@ func RunCmdRouter(f kcmdutil.Factory, cmd *cobra.Command, out, errout io.Writer,
 			},
 		},
 	}
+	if cfg.Type == "haproxy-router" && cfg.SyslogSidecar {
+		containers = append(containers, kapi.Container{
+			Name:  "syslog",
+			Image: cfg.ImageTemplate.ExpandOrDie("egress-dns-proxy"),
+			Command: []string{
+				"/sbin/rsyslogd", "-n",
+				// TODO: Once we have an image with rsyslog 8.32
+				// or later, we can switch to -i NONE.
+				"-i", "/tmp/rsyslog.pid",
+				"-f", "/etc/rsyslog/rsyslog.conf",
+			},
+			ImagePullPolicy: kapi.PullIfNotPresent,
+			VolumeMounts: []kapi.VolumeMount{
+				kapi.VolumeMount{
+					Name:      "rsyslog-config",
+					MountPath: "/etc/rsyslog",
+				},
+				kapi.VolumeMount{
+					Name:      "rsyslog-socket",
+					MountPath: "/var/lib/rsyslog",
+				},
+			},
+			Resources: kapi.ResourceRequirements{
+				Requests: kapi.ResourceList{
+					kapi.ResourceCPU:    resource.MustParse("100m"),
+					kapi.ResourceMemory: resource.MustParse("256Mi"),
```

Comment:
I don't know what reasonable defaults would be for rsyslog.  Do we even need to specify resource requirements for rsyslog?

Reply:
Not sure re: needing to specify the resource requirements but on the surface of it the defaults seem to be reasonable. 
256MiB seems fine as rsyslog uses a 10mb buffer per disk file/queue (and we have just the one or two), so probably more than enough for buffers + its own footprint.
10% of a core for seems ok though I think it would be depend on the amount of log messages processed.

Reply:
rsyslog shouldn't use any disk during operation—it's configured to read from a Unix domain socket and forward to standard output.  It may fill stdout's buffer, which should be capped at `/proc/sys/fs/pipe-max-size`, which defaults to 1 MiB.

rsyslog may use a lot of CPU if `ROUTER_LOG_LEVEL` is cranked up to get access logs (which was the original motivation for this change) and the router is seeing a lot of activity.

Reply:
I don't have a strong opinion but based on your above comment, don't we want to limit the cpu/memory usage of the rsyslog instead of setting minimum cpu/memory request for the sidecar ? 
May be this needs to be relative (percent instead of fixed) to how much is available for the router.

Reply:
> I don't have a strong opinion but based on your above comment, don't we want to limit the cpu/memory usage of the rsyslog instead of setting minimum cpu/memory request for the sidecar ?

All right, so delete `Requests:` and add `Limits:`, right?

> May be this needs to be relative (percent instead of fixed) to how much is available for the router.

The router container has hard-coded resource request values.  The pod spec does not specify resources.  Do you mean the syslog container's resource limits should be computed relative to the hard-coded defaults for the router container, or computed based on the host's CPU/memory? I don't see the point of the former, and we don't know the latter.